### ISSUE TITLE
Multiple files in $KUBECONFIG trigger "Error loading config file"

### ISSUE THREAD

**Issue by nea** [fission/fission#821]
Hi

* Platform Windows

I am using a multiple config file approach and have set up all files as semicolon separated list under `KUBECONFIG`. Now, fission says

```
Failed to connect to Kubernetes: Error loading config file
```

with the list of files as string. Of course, it cannot read it as it is not just one single file. Is this a known error? Are there workarounds Or do I have to merge all cluster configurations into one file?

Thanks a lot in advance

**Comment by vishal-biyani** [comment]
@nea Can you check if you can run `kubectl get pods` successfully?

**Comment by nea** [comment]
Hi @vishal-biyani 

Yeah, that was my first thought, too. But that works. For example:

```
NAME                              READY     STATUS    RESTARTS   AGE
buildermgr-d86d68dcf-rrrgj        1/1       Running   2          1h
controller-7fbcc99f54-s9mz6       1/1       Running   0          1h
executor-d869645ff-5cdch          1/1       Running   3          1h
influxdb-6799f75947-7p825         1/1       Running   0          1h
kubewatcher-5784b586f4-9f5zs      1/1       Running   2          1h
logger-5lx4t                      1/1       Running   0          1h
mqtrigger-57775c8766-p97cn        1/1       Running   3          1h
nats-streaming-7686cd7ff4-stbgk   1/1       Running   0          1h
router-669f5cf765-vgjsc           1/1       Running   2          1h
storagesvc-66c447d67-nbrmr        1/1       Running   0          1h
timer-678f7b8884-mp7kz            1/1       Running   2          1h
```

for ` kubectl get pods -n fission`. 

I also did some context switching as explained at the Kubernetes doc to verify that my two clusters are working.


Thanks

**Comment by vishal-biyani** [comment]
Thanks for confirming that, I think I am able to reproduce it the way you mentioned. For now I suggest either merge kubeconfigs into one or export only one file in Kubeconfig in that shell. I will leave this issue open so we can fix it later.

**Comment by nea** [comment]
Hey @vishal-biyani 

Thanks for the heads up. Will do until further notice.


Best

**Comment by soamvasani** [comment]
Ok, so I think are supposed to use [`ClientConfigLoadingRules`](https://godoc.org/k8s.io/client-go/tools/clientcmd#ClientConfigLoadingRules) with the `Precedence` field set to the list of paths in the KUBECONFIG.

This should replace the usage of `clientcmd.BuildConfigFromFlags` which only seems to support a single config file.

**Comment by life1347** [comment]
Solved by #1126
